### PR TITLE
getAssetRules method queries optimized

### PR DESCRIPTION
**Pull request by nadeeshaan**
Previous query is consisted with a join operation. In the changed query, first get the lft and rgt values related to the asset id or to the asset name. Then Another query is executed to get the rules.

### CODE REVIEW

**Comment by gunjanpatel** [comment]
@nadeeshaan Please don't forget to add Testing Instructions.

File: libraries/joomla/access/access.php [hunk 1]
```diff
@@ -220,35 +220,51 @@ public static function getAssetRules($asset, $recursive = false)
...
+			// Extract the rules from the result array
+			foreach ($result as $key => $value)
+			{
+				$temp[$key] = $value[0];
+				$i++;
+			}
+
+			$result = $temp;
+		}
 		// Get the root even if the asset is not found and in recursive mode
```
Comment (gunjanpatel) [inline libraries/joomla/access/access.php:275]: Add one empty space before line comment.

**Comment by nadeeshaan** [comment]
**Testing:**
    When you accessing a page in which articles are being showed this query is executed.

**File Location:**

 `JROOT \\libraries\joomla\access\access.php`

**Method:**

`getAssetRules($asset, $recursive = false)`

**Comment by alikon** [comment]
``` Sql
SELECT rules
  FROM new_assets
  WHERE (id = 1)
  GROUP BY id
```

 there is no need to group by when accessing by primary key
- the bug seems fixed

File: libraries/joomla/access/access.php [hunk 2]
```diff
@@ -219,35 +219,59 @@ public static function getAssetRules($asset, $recursive = false)
...
+			}
+			else
+			{
+				$query->where('(name = ' . $db->quote($asset) . ')');
+			}
 
-		// Execute the query and load the rules from the result.
-		$db->setQuery($query);
-		$result = $db->loadColumn();
+			$query->group('id');
```
Comment (alikon) [inline libraries/joomla/access/access.php]: when accessing by PK don't need to group by

**Comment by alikon** [comment]
send to joomlacms

**Comment by nadeeshaan** [comment]
## Testing Instructions:

In order to test this PR, first apply the PR via the Joomla! Patch Tester and then login to the Joomla! site. When accessing the pages where articles are being showed this query triggers.

Article Pages

### Original Query before the Optimization

``` Sql
SELECT b.rules
FROM #__assets AS a
LEFT JOIN #__assets AS b 
    ON b.lft <= a.lft 
    AND b.rgt >= a.rgt
WHERE (a.name = 'com_content')
GROUP BY b.id, b.rules, b.lft
ORDER BY b.lft
```

### After the Optimization

#### Decomposed Query 1:

``` sql
SELECT lft,rgt
FROM #__assets
WHERE (name = 'com_content')
```

#### Decomposed Query 2:

``` sql
SELECT rules,lft
FROM #__assets
WHERE lft <= retrieved_lft_value 
AND rgt >= retrieved_rgt_value
```

### Query Change Description

As we compare the previous query and the decomposed queries you can clearly see that the JOIN operation of the previous query has been removed not using the sub selects, but first retrieving the required `lft` and `rgt` values from the `#__assets` and then use these values normally to retrieve the rules values from the `#_assets` table using a simple select query

**Comment by gunjanpatel** [comment]
@nadeeshaan Is this also sent to joomla-cms?

**Comment by nadeeshaan** [comment]
@gunjanpatel all the PRs with the label "Sent to JCM" has been sent to joomla-cms

**Comment by gunjanpatel** [comment]
Thanks @nadeeshaan I will close all with label then. I have created task for you in out new repo. Please move all PRs to that repo.

**Pull request closed without merging**